pkg/vpcmodel/nodesConnectivity.go

```diff
@@ -188,19 +190,19 @@ func (v *VPCConnectivity) computeCombinedConnectionsPerDirection(isIngressDirect
 			otherDirectionConns := v.AllowedConns[peerNode].ingressOrEgressAllowedConns(!isIngressDirection)[node]
 			combinedConns = combinedConns.Intersect(otherDirectionConns)
 		}
-		v.AllowedConnsCombined.updateAllowedConnsMap(src, dst, combinedConns)
+		allowedConnsCombined.updateAllowedConnsMap(src, dst, combinedConns)
 	}
 }
 
 // computeAllowedConnsCombined computes combination of ingress&egress directions per connection allowed
-// the result for this computation is stateless connections
-// (could be that some of them or a subset of them are stateful,but this is not computed here)
-func (v *VPCConnectivity) computeAllowedConnsCombined() {
-	v.AllowedConnsCombined = GeneralConnectivityMap{}
+// the stateful state of the connectivity is not computed here
+func (v *VPCConnectivity) computeAllowedConnsCombined() GeneralConnectivityMap {
+	allowedConnsCombined := GeneralConnectivityMap{}
 	for node, connectivityRes := range v.AllowedConns {
-		v.computeCombinedConnectionsPerDirection(true, node, connectivityRes)
-		v.computeCombinedConnectionsPerDirection(false, node, connectivityRes)
+		v.computeCombinedConnectionsPerDirection(true, node, connectivityRes, allowedConnsCombined)
+		v.computeCombinedConnectionsPerDirection(false, node, connectivityRes, allowedConnsCombined)
 	}
```

Comment:
create GeneralConnectivityMap{} inside computeCombinedConnectionsPerDirection(), return it as an output, and use
updateMap() to merge the two outputs

Reply:
Not sure its a good idea. In the worst case it will double the complexity as each edge will be accessed twice: once for computing it and once for adding it.